sqsdomain/cw_pool.go

```diff
@@ -80,3 +96,94 @@ type TransmuterAssetConfig struct {
 	// [more info](https://github.com/osmosis-labs/transmuter/tree/v3.0.0?tab=readme-ov-file#normalization-factors)
 	NormalizationFactor osmomath.Int `json:"normalization_factor"`
 }
+
+type OrderbookDirection int
+
+const (
+	BID OrderbookDirection = 1
+	ASK OrderbookDirection = -1
+)
+
+func (d *OrderbookDirection) String() string {
+	switch *d {
+	case BID:
+		return "BID"
+	case ASK:
+		return "ASK"
+	default:
+		return "UNKNOWN"
+	}
+}
+
+func (d *OrderbookDirection) Opposite() OrderbookDirection {
+	switch *d {
+	case BID:
+		return ASK
+	case ASK:
+		return BID
+	default:
+		return 0
+	}
+}
+
+// OrderbookData, since v1.0.0
+type OrderbookData struct {
+	QuoteDenom  string           `json:"quote_denom"`
+	BaseDenom   string           `json:"base_denom"`
+	NextBidTick int64            `json:"next_bid_tick"`
+	NextAskTick int64            `json:"next_ask_tick"`
+	Ticks       []TickIdAndState `json:"ticks"`
+}
+
+// Returns tick state index for the given ID
+func (d *OrderbookData) GetTickIndexById(tickId int64) int {
+	for i, tick := range d.Ticks {
+		if tick.TickId == tickId {
+			return i
+		}
+	}
+	return -1
+}
+
+type TickValues struct {
+	// Total Amount of Liquidity at tick (TAL)
+	// - Every limit order placement increments this value.
+	// - Every swap at this tick decrements this value.
+	// - Every cancellation decrements this value.
+	TotalAmountOfLiquidity osmomath.BigDec `json:"total_amount_of_liquidity"`
+}
```

Comment:
Why does this have to be in a struct and not as `BigDec` directly?

Having this in a struct introduces light serialization overhead

Reply:
agreed, the original `TickValues` has a lot more field, this is all we need to sqs. Will remove the wrapped struct.